test/index.js

```diff
@@ -1073,5 +1109,117 @@ var HasManyGetter = require('../services/has-many-getter');
+                    done();
+                  });
+              })
+              .catch(done);
+          });
+        });
+
+        describe('On BelongsToMany relationship', function () {
+          it('should delete the relationship of the record', function (done) {
+            var params = {
+              recordId: '100',
+              associationName: 'teams'
+            };
+            var data = {
+              data: [
+                { id: '100', type: 'team' }
+              ]
+            };
+            return new HasManyDissociator(models.user, models.team,
+              sequelizeOptions, params, data)
+              .perform()
+              .then(function () {
+                return models.userTeam
+                  .find({ where: { userId: '100', teamId: '100' } })
+                  .then(function (userTeam) {
+                    expect(userTeam).to.be.null;
+                    done();
+                  });
+              })
+              .catch(done);
+          });
+        });
+      });
+
+      describe('Delete', function () {
+        describe('On HasMany relationship', function () {
+          it('should delete the relationship and delete the record', function (done) {
+            var params = {
+              recordId: '100',
+              associationName: 'addresses',
+              delete: 'true'
+            };
+            var data = {
+              data: [
+                { id: '103', type: 'address' }
+              ]
+            };
+            return new HasManyDissociator(models.user, models.address,
+              sequelizeOptions, params, data)
+              .perform()
+              .then(function () {
+                return models.address
+                  .find({ where: { id: '103' } })
+                  .then(function (address) {
+                    expect(address).to.be.null;
+                    done();
+                  });
+              })
+              .catch(done);
+          });
+        });
+
+        describe('On BelongsToMany relationship', function () {
+          it('should delete the relationship and delete the record', function (done) {
+            var params = {
+              recordId: '100',
+              associationName: 'teams',
+              delete: 'true'
+            };
+            var data = {
+              data: [
+                { id: '100', type: 'team' }
+              ]
+            };
+            return new HasManyDissociator(models.user, models.team,
+              sequelizeOptions, params, data)
+              .perform()
+              .then(function () {
+                return models.userTeam
+                  .find({ where: { userId: '100', teamId: '100' } })
+                  .then(function (userTeam) {
+                    expect(userTeam).to.be.null;
+                    done();
+                  });
+              })
+              .catch(done);
+          });
+        });
+      });
+    });
```

Comment:
You can add tests which try to delete and dissociate on a record which is not an relationship.